src/generateNotes.ts

```diff
@@ -1,148 +1,49 @@
+  const platformVersions: { [key in keyof PlatformConfig]?: string } = {}
+
+  // Normalize plugin config: prefer `platforms`, fall back to legacy top-level platform keys for backward compatibility
+  // TODO: Remove support for top-level `android` and `iOS` keys in the next major release (BC)
+  const platforms: PlatformConfig = config.platforms || {}
+  if (!config.platforms) {
+    if (config.android) {
+      ctx.logger.warn('[DEPRECATED] Use `platforms.android` instead of top-level `android` in plugin config.')
+      platforms.android = config.android
     }
-    if (child.stderr === null) {
-      reject(new Error('Unexpected error: stderr of subprocess is null'))
-      return
-    }
-
-    let androidVersion: string | null = null
-    child.stdout.on('data', (line: Buffer) => {
-      if (!line || line.toString().trim() === '') {
-        return
-      }
-
-      logger.debug(`Gradle stdout: ${line}`)
-      androidVersion = line.toString().trim()
-    })
-    child.stderr.on('data', (line: Buffer) => {
-      logger.error(line.toString().trim())
-    })
-    child.on('close', (code: number) => {
-      if (code !== 0) {
-        reject(new Error(`Unexpected error: Gradle failed with status code ${code}`))
-        return
-      }
-
-      if (androidVersion === null) {
-        reject(new Error(`Could not read output of \`${androidGradleTaskName}\` gradle task.`))
-        return
-      }
-
-      resolve(androidVersion)
-    })
-    child.on('error', (err) => {
-      logger.error(err)
-      reject(err)
-    })
-  })
-}
-
-type PodspecJson = {
-  dependencies: {
-    [key: string]: [string]
-  }
-}
-
-export const getIOSVersion = async (cwd: string, iOSPodSpecJsonPath: string, dependencyName: string) => {
-  const jsonFile = join(cwd, iOSPodSpecJsonPath)
 
-  let fileContent: string
-  try {
-    fileContent = readFileSync(jsonFile, 'utf8')
-  } catch (error: any) {
-    switch (error.code) {
-      case 'ENOENT':
-        throw new Error(`${iOSPodSpecJsonPath} file does not exist.`)
-      case 'EACCES':
-        throw new Error(`${iOSPodSpecJsonPath} file cannot be accessed.`)
-      default:
-        throw new Error(`${iOSPodSpecJsonPath} file cannot be read.`)
+    if (config.iOS) {
+      ctx.logger.warn('[DEPRECATED] Use `platforms.iOS` instead of top-level `iOS` in plugin config.')
+      platforms.iOS = config.iOS
     }
   }
 
-  let data: PodspecJson
-  try {
-    data = JSON.parse(fileContent) as PodspecJson
-  } catch (error) {
-    throw new Error(`${iOSPodSpecJsonPath} file cannot be parsed as JSON.`)
+  if (!platforms.android && !platforms.iOS) {
+    throw new Error('No platforms specified. You must configure at least one platform under `platforms`.')
   }
 
-  if (!data.dependencies || !data.dependencies[dependencyName]) {
-    throw new Error(`${iOSPodSpecJsonPath} file does not contain '${dependencyName}' in dependencies.`)
+  if (platforms['android']) {
```

Comment:
nit: is there a reason behind using bracket notation here?

Reply:
Updated with commit [#cc1f704](https://github.com/fingerprintjs/semantic-release-native-dependency-plugin/pull/3/commits/cc1f704efb767aec09250edf0df1c68186f62258)